### PR TITLE
[11.0] [MIG] account_payment_pain_base

### PR DESCRIPTION
**Pull request by etobella**
Migration to 11.0.
Tests should be performed with account_banking_sepa_credit_transfer and account_banking_sepa_direct_debit.
Depends on:
 - [x] account_payment_mode #399
 - [x] account_payment_partner #400
 - [x] account_payment_order #401

### CODE REVIEW

File: account_banking_pain_base/README.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+This module depends on :
+
+- account_payment_order
+
+This module is part of the OCA/bank-payment suite.
+
+Configuration
+=============
+
+#. Go to Accounting > Configuration > Settings.
```
Comment (rgarnau) [inline account_banking_pain_base/README.rst]: Now 'Accounting' is 'Invoicing'.

File: account_banking_pain_base/README.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+#. On the fields "Initiating Party Issuer" and "Initiating Party Identifier",
+   in the section *SEPA/PAIN*, you can fill the corresponding identifiers.
+
+If your country requires several identifiers (like Spain), you must:
+
+#. Go to *Accounting > Configuration > Settings*.
```
Comment (rgarnau) [inline account_banking_pain_base/README.rst]: Invoicing

File: account_banking_pain_base/README.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+#. On the section *SEPA/PAIN*, check the mark "Multiple identifiers".
+#. Now go to *Accounting > Configuration > Management > Payment Modes*.
```
Comment (rgarnau) [inline account_banking_pain_base/README.rst]: Invoicing

File: account_banking_pain_base/models/res_company.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    initiating_party_issuer = fields.Char(
+        string='Initiating Party Issuer', size=35,
+        help="This will be used as the 'Initiating Party Issuer' in the "
+        "PAIN files generated by Odoo.")
+    initiating_party_identifier = fields.Char(
+        string='Initiating Party Identifier', size=35,
+        help="This will be used as the 'Initiating Party Identifier' in "
+        "the PAIN files generated by Odoo.")
+
+    @api.one
```
Comment (rgarnau) [inline account_banking_pain_base/models/res_company.py]: deprecated

**Review by rgarnau — COMMENTED** [review]
LGTM :+1: (minor comments inline)

**Review by rgarnau — APPROVED** [review]

**Review by HviorForgeFlow — APPROVED** [review]
LGTM

File: account_banking_pain_base/models/account_config_settings.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Akretion - Alexis de Lattre <alexis.delattre@akretion.com>
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
```
Comment (astirpe) [inline account_banking_pain_base/models/account_config_settings.py]: The file `account_config_settings.py` should be renamed to `res_config_settings.py`

File: account_banking_pain_base/models/res_company.py [hunk 6]
```diff
@@ -22,7 +22,6 @@ class ResCompany(models.Model):
         help="This will be used as the 'Initiating Party Identifier' in "
         "the PAIN files generated by Odoo.")
 
-    @api.one
```
Comment (astirpe) [inline account_banking_pain_base/models/res_company.py]: It is ok to remove `@api.one` here, but mind that `self` is supposed to be a singleton.

In my opinion you should add a `self.ensure_one()` in `method _default_initiating_party()`, and then, in file `post_install.py`, ensure that the method is invoked separately for each company. That means, replace:

```
companies._default_initiating_party()
```

with:

```
for company in companies:
    company._default_initiating_party()
```

**Review by astirpe — CHANGES_REQUESTED** [review]

**Comment by angelmoya** [comment]
Functional tested OK :+1:

**Comment by etobella** [comment]
@astirpe changes applied.

**Review by astirpe — APPROVED** [review]

**Pull request merged**